components/brave_extension/extension/brave_extension/containers/braveShields.tsx

```diff
@@ -87,6 +88,19 @@ export default class Shields extends React.PureComponent<Props, State> {
     this.props.actions.shieldsReady()
   }
 
+  componentDidUpdate (prevProps: Props) {
+    // If current URL does not match current URL data in Shields,
+    // close it immediately.
+    // See https://github.com/brave/brave-browser/issues/6601.
+    const previousURL: string = prevProps.shieldsPanelTabData.url
+    const currentURL: string = this.props.shieldsPanelTabData.url
+    if (previousURL !== currentURL) {
+      if (shieldsHasFocus(currentURL) === false) {
```

Comment:
How about we just close the popup when the browser window is not focused?
As a user I would think that makes more sense and consistent to me, but I'm fine with the current approach too.
cc @karenkliu for which one we should go with.

Reply:
I agree, updated to cover all cases.